Runtime/Utilities/Texture2DUtility.cs

```diff
@@ -0,0 +1,300 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace StansAssets.Foundation
+{
+    /// <summary>
+    /// Texture2D Utility methods.
+    /// </summary>
+    public class Texture2DUtility
+    {
+        /// <summary>
+        /// Texture scale mode options.
+        /// </summary>
+        public enum TextureScaleMode
+        {
+            /// <summary>
+            /// Nearest.
+            /// </summary>
+            Nearest = 0,
+
+            /// <summary>
+            /// Bilinear
+            /// </summary>
+            Bilinear = 1,
+
+            /// <summary>
+            /// Average
+            /// </summary>
+            Average = 2,
+        }
+
+        static readonly Dictionary<float, Texture2D> s_ColorIcons = new Dictionary<float, Texture2D>();
+
+        /// <summary>
+        /// Generates plane color <see cref="Texture2D"/> object with given size.
+        /// You can also choose if you would like to cache generated image, so the next time you ask for the image with the same color,
+        /// cache result will be returned instead of generating new image.
+        /// </summary>
+        /// <param name="colorHtmlString">
+        /// Case insensitive html string to be converted into a color.
+        /// <see cref="ColorHelper.MakeColorFromHtml(string)"/> for more info about <see cref="colorHtmlString"/> format.
+        /// </param>
+        /// <param name="width"> Texture width.</param>
+        /// <param name="height"> Texture height.</param>
+        /// <param name="cacheGeneratedImage">When set to `true` the generated image is cached.</param>
+        public static Texture2D MakePlainColorImage(string colorHtmlString, int width = 1, int height = 1, bool cacheGeneratedImage = true)
+        {
+            var color = ColorHelper.MakeColorFromHtml(colorHtmlString);
```

Comment:
Looks like we can replace `ColorHelper.MakeColorFromHtml` with `ColorUtility.TryParseHtmlString`
```suggestion
            ColorUtility.TryParseHtmlString(colorHtmlString, out var color);
```